File: benchmarks/python/16_array_of_objects.py
class Particle:
    __slots__ = ('x', 'y', 'mass')
    def __init__(self, x, y, mass):
        self.x = x
        self.y = y
        self.mass = mass

def array_of_objects(n):
    particles = []
    for i in range(n):
        particles.append(Particle(i * 1.0, i * 2.0, i * 0.5))
    total_mass = 0.0
    for i in range(n):
        total_mass += particles[i].mass
    return total_mass
print(array_of_objects(100000))

File: benchmarks/node/16_array_of_objects.mjs
function array_of_objects(n) {
    const particles = [];
    for (let i = 0; i < n; i++) {
        particles.push({ x: i * 1.0, y: i * 2.0, mass: i * 0.5 });
    }
    let total_mass = 0;
    for (let i = 0; i < n; i++) {
        total_mass += particles[i].mass;
    }
    return total_mass;
}
console.log(array_of_objects(100000));
